pkg/kubelet/cm/devicemanager/plugin/v1beta1/server.go

```diff
@@ -97,20 +98,44 @@ func (s *server) Start() error {
 		return err
 	}
 
-	ln, err := net.Listen("unix", s.SocketPath())
-	if err != nil {
-		klog.ErrorS(err, "Failed to listen to socket while starting device plugin registry")
-		return err
-	}
-
 	s.wg.Add(1)
-	s.grpc = grpc.NewServer([]grpc.ServerOption{}...)
-
-	api.RegisterRegistrationServer(s.grpc, s)
+	maxRetries := 5
```

Comment:
why we need max retries? Why is constant defined inside the function?

Reply:
workshopping how to prevent indefinite retries

Reply:
why would we need to prevent indefinite retries? With the proper backoff policy we should never stop (before the configured healthy checks will kill kubelet)